tokio/Cargo.toml

```diff
@@ -51,20 +51,19 @@ macros = ["tokio-macros"]
 stats = []
 net = [
   "libc",
+  "mio/net",
+  "mio/os-ext",
   "mio/os-poll",
-  "mio/os-util",
-  "mio/tcp",
-  "mio/udp",
-  "mio/uds",
+  "socket2/all",
```

Comment:
I'm not too familiar with the features of socket2, but can we be more precise here?

Reply:
socket2's `all` feature means enabling APIs that are not available in some OSes, not means enabling all feature flags.

https://github.com/rust-lang/socket2/blob/6601ed132b37d6e9d178b34918bfb0b236800232/Cargo.toml#L42-L43

> ```toml
> # Enable all API, even ones not available on all OSs.
> all = []
> ```

In our case, this feature is needed to use [reuse_port](https://docs.rs/socket2/0.4.2/socket2/struct.Socket.html#method.reuse_port)/[set_reuse_port](https://docs.rs/socket2/0.4.2/socket2/struct.Socket.html#method.set_reuse_port), which is not available in Solaris & Illumos.